src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_RANDOM_VALUE = 10;
+    public static final String DEFAULT_COLOR = "WHITE";
+    public static final double DEFAULT_RADIUS = 10;
+
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                return new Square(color, random.nextInt(MAX_RANDOM_VALUE) + 1);
+            case 1:
+                return new Rectangle(color, random.nextInt(MAX_RANDOM_VALUE) + 1,
+                        random.nextInt(MAX_RANDOM_VALUE) + 1);

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables for width and length and then pass them on for better code readability.